### ISSUE TITLE
Nullable MorphTo relation

### ISSUE THREAD

**Issue by Livijn** [laravel/nova-issues#423]
I am trying to allow for a nullable relation, but it is not working. Here is the field:
```
MorphTo::make('Awardable')->types([
                    User::class,
                    Team::class,
                ])->rules('nullable'),
```

This is the result:
![screenshot](https://i.gyazo.com/2264cae8e4e2a792591b37d1eac36999.png)

**Comment by BrianLangevin** [comment]
I ended up solving the same issue by extending the MorphTo field class and overriding the getRules() and fill() methods. 


```
class NullableMorphTo extends MorphTo
{
	/**
	 * Strip the 'required' rules from the rules array.
	 *
	 * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
	 * @return array
	 */
	public function getRules(NovaRequest $request)
	{
		$rules = parent::getRules($request);

		foreach ($rules as $attribute => $value) {
			if (in_array('required', $value)) {
				unset($rules[$attribute][array_search('required', $value)]);
			}
		}

		return $rules;
	}

	/**
	 * Do not hydrate the model if no relation was set.
	 *
	 * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
	 * @param  object  $model
	 * @return void
	 */
	public function fill(NovaRequest $request, $model)
	{
		if (! $request->has($this->attribute)) {
			return;
		}

		parent::fill($request, $model);
	}
}

```

**Comment by FrittenKeeZ** [comment]
@BrianLangevin your `getRules()` can be simplified to just:
```php
/**
 * Strip the 'required' rules from the rules array.
 *
 * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
 * @return array
 */
public function getRules(NovaRequest $request)
{
    return array_diff(parent::getRules($request), ['required']);
}
```

**Comment by Livijn** [comment]
@alexbowers Why close this issue when it's still unresolved?

**Comment by Livijn** [comment]
Well, solving the issue by extending the original `MorphTo` class is more of a work around.

From the [docs](https://nova.laravel.com/docs/1.0/resources/validation.html#creation-rules): 
```
If you would like to define rules that only apply when a resource is being created, you may use the creationRules method:
```

If the `nullable` rule is intended to work, then it is still an issue. Maybe it is not implemented or not intended to be implemented, then I guess we will have to use our own field class.

**Comment by alexbowers** [comment]
Fair enough, reopened.

**Comment by davidhemphill** [comment]
I believe this was fixed in 1.1.7. 👍
